designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala

```diff
@@ -37,6 +40,18 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
           modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
         }
       }
+    } ~ path("processDefinitionData" / "categoriesWithProcessingType") {
+      get {
+        complete {
+          val categories = user match {
+            case user: CommonUser => processCategoryService.getAllCategories.filter(user.can(_, Read))
+            case _: AdminUser => processCategoryService.getAllCategories
+          }
+          categories
+            .map(category => category -> processCategoryService.getTypeForCategory(category))
+            .toMap[Category, Option[ProcessingType]]
```

Comment:
is it possible, that there is no ProcessingType (None) for a category?

Reply:
API defines `getTypeForCategory` with `Option[ProcessingType]` so I didn't try to go deep into it. As I look at it now I think that in my case it always have processingType but I can see cases when this `getTypeForCategory` can return None. 

We could drop Option but it wouldn't make any difference. WDYT?

Reply:
ok lets leave it